Formula/ntopng.rb

```diff
@@ -1,9 +1,18 @@
 class Ntopng < Formula
   desc "Next generation version of the original ntop"
   homepage "http://www.ntop.org/products/ntop/"
-  url "https://downloads.sourceforge.net/project/ntop/ntopng/ntopng-2.2.tar.gz"
-  sha256 "4fccfc9e9f333addcd3c957b4520c471117bc2df5655d6eabf328c7385fb255e"
-  revision 1
+
+  stable do
+    url "https://github.com/ntop/ntopng/archive/2.4.tar.gz"
+    sha256 "86f8ed46983f46bcd931304d3d992fc1af572b11e461ab9fb4f0f472429bd5dd"
+
+    resource "nDPI" do
+      # tip of 1.8-stable branch; four commits beyond the 1.8 tag
+      url "https://github.com/ntop/nDPI.git",
+        :tag => "1.8-stable",
```

Comment:
Being able to specify a branch with `:tag =>` sounds like a bug and should probably be fixed [in `GitDownloadStrategy#checkout`](https://github.com/Homebrew/brew/blob/3c5f007bfa1a469ef3e1a048ee163f3eaa73cb71/Library/Homebrew/download_strategy.rb#L679-L682) by using a fully qualified reference like `"refs/heads/#{@ref}"` or `"refs/tags/#{@ref}"` instead of just `@ref`.

Using `:branch` is probably designed to actually track a moving target, thus checking the `:revision` in this case doesn't make much sense ([and thus isn't done](https://github.com/Homebrew/brew/blob/3c5f007bfa1a469ef3e1a048ee163f3eaa73cb71/Library/Homebrew/download_strategy.rb#L138-L145)). Printing a warning or even generating an error if `:revision` is specified nonetheless or simply always checking the revision if it is specified sound like possible solutions to me.

What do you think?